erreur sur un des filtres avancés

**Version 2.9.2 mais aussi 2.10.1 de la demo**

**Description du bug**
Le filtre avancé statut biogéographique" dans la synthèse, GN (2.9.2) remonte une erreur. 

**Comment reproduire**
Choisir le filtre avancé "statut biogéographique" sur [demo.geonature.fr/geonature](http://demo.geonature.fr/geonature) et lancer la recherche

**Logs**
```
Traceback (most recent call last):
  File "/home/geonature/geonature/backend/venv/lib/python3.9/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/geonature/geonature/backend/venv/lib/python3.9/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/geonature/geonature/backend/geonature/core/gn_permissions/decorators.py", line 82, in __check_cruved_scope
    return fn(*args, **kwargs)
  File "/home/geonature/geonature/backend/venv/lib/python3.9/site-packages/utils_flask_sqla/response.py", line 19, in _json_resp
    res = fn(*args, **kwargs)
  File "/home/geonature/geonature/backend/geonature/core/gn_synthese/routes.py", line 151, in get_observations_for_web
    synthese_query_class.filter_query_all_filters(info_role)
  File "/home/geonature/geonature/backend/geonature/core/gn_synthese/utils/query_select_sqla.py", line 371, in filter_query_all_filters
    self.filter_other_filters()
  File "/home/geonature/geonature/backend/geonature/core/gn_synthese/utils/query_select_sqla.py", line 340, in filter_other_filters
    col = getattr(self.model.__table__.columns, colname)
  File "/home/geonature/geonature/backend/venv/lib/python3.9/site-packages/sqlalchemy/util/_collections.py", line 212, in __getattr__
    raise AttributeError(key)
AttributeError: id_nomenclature_biogeo_status
[2022-12-15 15:46:50 +0100] [993832] [ERROR] Exception on /synthese/for_web [POST]
Traceback (most recent call last):
  File "/home/geonature/geonature/backend/venv/lib/python3.9/site-packages/sqlalchemy/util/_collections.py", line 210, in __getattr__
    return self._data[key]
KeyError: 'id_nomenclature_biogeo_status'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/geonature/geonature/backend/venv/lib/python3.9/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/geonature/geonature/backend/venv/lib/python3.9/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/geonature/geonature/backend/geonature/core/gn_permissions/decorators.py", line 82, in __check_cruved_scope
    return fn(*args, **kwargs)
  File "/home/geonature/geonature/backend/venv/lib/python3.9/site-packages/utils_flask_sqla/response.py", line 19, in _json_resp
    res = fn(*args, **kwargs)
  File "/home/geonature/geonature/backend/geonature/core/gn_synthese/routes.py", line 151, in get_observations_for_web
    synthese_query_class.filter_query_all_filters(info_role)
  File "/home/geonature/geonature/backend/geonature/core/gn_synthese/utils/query_select_sqla.py", line 371, in filter_query_all_filters
    self.filter_other_filters()
  File "/home/geonature/geonature/backend/geonature/core/gn_synthese/utils/query_select_sqla.py", line 340, in filter_other_filters
    col = getattr(self.model.__table__.columns, colname)
  File "/home/geonature/geonature/backend/venv/lib/python3.9/site-packages/sqlalchemy/util/_collections.py", line 212, in __getattr__
    raise AttributeError(key)
AttributeError: id_nomenclature_biogeo_status
```

--- a/backend/geonature/core/gn_synthese/models.py
+++ b/backend/geonature/core/gn_synthese/models.py
@@ -668,6 +668,7 @@ class VSyntheseForWebApp(DB.Model):
     id_nomenclature_source_status = DB.Column(DB.Integer)
     id_nomenclature_determination_method = DB.Column(DB.Integer)
     id_nomenclature_behaviour = DB.Column(DB.Integer)
+    id_nomenclature_biogeo_status = DB.Column(DB.Integer)
     reference_biblio = DB.Column(DB.Unicode)
     name_source = DB.Column(DB.Unicode)
     url_source = DB.Column(DB.Unicode)

--- a/backend/geonature/migrations/versions/cb663f039774_add_biogeo_status_v_synthese.py
+++ b/backend/geonature/migrations/versions/cb663f039774_add_biogeo_status_v_synthese.py
@@ -1,0 +1,205 @@
+"""[synthese] add biogeo status
+
+Revision ID: cb663f039774
+Revises: a90fa0630ef5
+Create Date: 2026-09-04 15:21:05.413864
+
+"""
+
+from alembic import op
+import sqlalchemy as sa
+from utils_flask_sqla.migrations.utils import logger
+
+# revision identifiers, used by Alembic.
+revision = "cb663f039774"
+down_revision = "a90fa0630ef5"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.execute(
+        """
+        CREATE OR REPLACE VIEW gn_synthese.v_synthese_for_web_app AS
+        SELECT s.id_synthese,
+            s.unique_id_sinp,
+            s.unique_id_sinp_grp,
+            s.id_source,
+            s.entity_source_pk_value,
+            s.count_min,
+            s.count_max,
+            s.nom_cite,
+            s.meta_v_taxref,
+            s.sample_number_proof,
+            s.digital_proof,
+            s.non_digital_proof,
+            s.altitude_min,
+            s.altitude_max,
+            s.depth_min,
+            s.depth_max,
+            s.place_name,
+            s."precision",
+            s.the_geom_4326,
+            st_asgeojson(s.the_geom_4326) AS st_asgeojson,
+            s.date_min,
+            s.date_max,
+            s.validator,
+            s.validation_comment,
+            s.observers,
+            s.id_digitiser,
+            s.determiner,
+            s.comment_context,
+            s.comment_description,
+            s.meta_validation_date,
+            s.meta_create_date,
+            s.meta_update_date,
+            s.last_action,
+            d.id_dataset,
+            d.dataset_name,
+            d.id_acquisition_framework,
+            s.id_nomenclature_geo_object_nature,
+            s.id_nomenclature_info_geo_type,
+            s.id_nomenclature_grp_typ,
+            s.grp_method,
+            s.id_nomenclature_obs_technique,
+            s.id_nomenclature_bio_status,
+            s.id_nomenclature_bio_condition,
+            s.id_nomenclature_naturalness,
+            s.id_nomenclature_exist_proof,
+            s.id_nomenclature_valid_status,
+            s.id_nomenclature_diffusion_level,
+            s.id_nomenclature_life_stage,
+            s.id_nomenclature_sex,
+            s.id_nomenclature_obj_count,
+            s.id_nomenclature_type_count,
+            s.id_nomenclature_sensitivity,
+            s.id_nomenclature_observation_status,
+            s.id_nomenclature_blurring,
+            s.id_nomenclature_source_status,
+            s.id_nomenclature_determination_method,
+            s.id_nomenclature_behaviour,
+            s.reference_biblio,
+            sources.name_source,
+            sources.url_source,
+            t.cd_nom,
+            t.cd_ref,
+            t.nom_valide,
+            t.lb_nom,
+            t.nom_vern,
+            s.id_module,
+            t.group1_inpn,
+            t.group2_inpn,
+            t.group3_inpn,
+            s.id_import,
+            t.id_statut,
+            t.id_rang,
+            t.regne,
+            t.phylum,
+            t.classe,
+            t.ordre,
+            t.famille,
+            t.sous_famille,
+            id_nomenclature_biogeo_status
+        FROM gn_synthese.synthese s
+            JOIN taxonomie.taxref t ON t.cd_nom = s.cd_nom
+            JOIN gn_meta.t_datasets d ON d.id_dataset = s.id_dataset
+            JOIN gn_synthese.t_sources sources ON sources.id_source = s.id_source;
+        """
+    )
+
+
+def downgrade():
+    logger.warning(
+        "La colonne id_nomenclature_biogeo_status ne peut pas être supprimée de la vue "
+        "gn_synthese.v_synthese_for_web_app sans la recréer (DROP + CREATE), ce qui pourrait casser "
+        "les vues dépendantes (notamment gn_dashboard.vm_taxonomie). La colonne est conservée "
+        "mais sa valeur remplacée par NULL."
+    )
+    op.execute(
+        """
+        CREATE OR REPLACE VIEW gn_synthese.v_synthese_for_web_app AS
+        SELECT s.id_synthese,
+            s.unique_id_sinp,
+            s.unique_id_sinp_grp,
+            s.id_source,
+            s.entity_source_pk_value,
+            s.count_min,
+            s.count_max,
+            s.nom_cite,
+            s.meta_v_taxref,
+            s.sample_number_proof,
+            s.digital_proof,
+            s.non_digital_proof,
+            s.altitude_min,
+            s.altitude_max,
+            s.depth_min,
+            s.depth_max,
+            s.place_name,
+            s."precision",
+            s.the_geom_4326,
+            st_asgeojson(s.the_geom_4326) AS st_asgeojson,
+            s.date_min,
+            s.date_max,
+            s.validator,
+            s.validation_comment,
+            s.observers,
+            s.id_digitiser,
+            s.determiner,
+            s.comment_context,
+            s.comment_description,
+            s.meta_validation_date,
+            s.meta_create_date,
+            s.meta_update_date,
+            s.last_action,
+            d.id_dataset,
+            d.dataset_name,
+            d.id_acquisition_framework,
+            s.id_nomenclature_geo_object_nature,
+            s.id_nomenclature_info_geo_type,
+            s.id_nomenclature_grp_typ,
+            s.grp_method,
+            s.id_nomenclature_obs_technique,
+            s.id_nomenclature_bio_status,
+            s.id_nomenclature_bio_condition,
+            s.id_nomenclature_naturalness,
+            s.id_nomenclature_exist_proof,
+            s.id_nomenclature_valid_status,
+            s.id_nomenclature_diffusion_level,
+            s.id_nomenclature_life_stage,
+            s.id_nomenclature_sex,
+            s.id_nomenclature_obj_count,
+            s.id_nomenclature_type_count,
+            s.id_nomenclature_sensitivity,
+            s.id_nomenclature_observation_status,
+            s.id_nomenclature_blurring,
+            s.id_nomenclature_source_status,
+            s.id_nomenclature_determination_method,
+            s.id_nomenclature_behaviour,
+            s.reference_biblio,
+            sources.name_source,
+            sources.url_source,
+            t.cd_nom,
+            t.cd_ref,
+            t.nom_valide,
+            t.lb_nom,
+            t.nom_vern,
+            s.id_module,
+            t.group1_inpn,
+            t.group2_inpn,
+            t.group3_inpn,
+            s.id_import,
+            t.id_statut,
+            t.id_rang,
+            t.regne,
+            t.phylum,
+            t.classe,
+            t.ordre,
+            t.famille,
+            t.sous_famille,
+            NULL::integer AS id_nomenclature_biogeo_status
+        FROM gn_synthese.synthese s
+            JOIN taxonomie.taxref t ON t.cd_nom = s.cd_nom
+            JOIN gn_meta.t_datasets d ON d.id_dataset = s.id_dataset
+            JOIN gn_synthese.t_sources sources ON sources.id_source = s.id_source;
+        """
+    )

